sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -2867,6 +2867,25 @@ pi_result cuda_piEnqueueKernelLaunch(
       retImplEv->start();
     }
 
+    // Set local mem max size if env var is present
+    static const char *local_mem_sz_ptr =
+        std::getenv("SYCL_PI_CUDA_MAX_LOCAL_MEM_SIZE");
+
+    if (local_mem_sz_ptr) {
+      int device_max_local_mem = 0;
+      cuDeviceGetAttribute(
+          &device_max_local_mem,
+          CU_DEVICE_ATTRIBUTE_MAX_SHARED_MEMORY_PER_BLOCK_OPTIN,
+          command_queue->get_context()->get_device()->get());
```

Comment:
```suggestion
          command_queue->device_->get());
```
Or alternatively add a `get_device` to `_pi_queue` in pi_cuda.hpp. The latter would be nicer, but I will leave it up to you.